### PR TITLE
Change the default delay before re-compilation from 750 to 1500 ms

### PR DESCRIPTION
**Pull request by OfekShilon**
Following some high-traffic alerts an idea was raised on discord to increase the default delay between typing and re-compilation - as today every slight hesitation triggers re-compilation.

The default delay was 750ms. This PR doubles it while trying to limit the change to users who didn't set it explicitly. On first navigation after this fix *all* 750's would be converted to 1500, but if a user later sets the delay to 750 it *won't* be converted (with a little help from `localStorage`).

This also sets a single place to centrally control the default delay in the future (personally I think it can be higher, but it can wait).

A new checkbox in the settings is added, and future modifications to the default delay would apply only when this is checked:
autodelay

### CODE REVIEW

File: static/settings.ts [hunk 1]
```diff
@@ -229,6 +235,8 @@ export class Settings {
         this.fillColourSchemeSelector(this.root.find('.colourScheme'), this.settings.theme);
         this.setSettings(this.settings);
         this.handleThemes();
+
+        this.eventHub.on('settingsChange', this.onSettingsChange.bind(this));
```
Comment (RubenRBS) [inline static/settings.ts:239]: This is now needed to be able to disable the slider based on the checkbox right? What was the behaviour without this line?

**Review by RubenRBS — COMMENTED** [review]
Nice! A few comments, otherwise looks good :)

File: static/settings.ts (hunk 1 above)
Comment (OfekShilon) [reply static/settings.ts:239]: Exactly - without this onSettingsChange was called only during `Settings` construction. I didn't look too closely, just matched the usage for other `onSettingsChange`s.

Comment (OfekShilon) [reply static/settings.ts:239]: This is certainly doable without subscribing to the event, just wanted to use this chance to make `onSettingsChange` used in the uniform way.

**Comment by OfekShilon** [comment]
Probably best to wait with this until the end of St. Louis (June 29)

**Comment by mattgodbolt** [comment]
I'm happy to try merging beforehand; this shouldn't really affect anything those folks would care about (unlike all the msvc nonsense we're hacking on too :D )

**Review by mattgodbolt — APPROVED** [review]

**Comment by mattgodbolt** [comment]
This is now live :)

**Pull request merged**